cdap-api/src/main/java/co/cask/cdap/api/workflow/AbstractWorkflow.java

```diff
@@ -189,4 +190,21 @@ protected final void createLocalDataset(String datasetName, Class<? extends Data
   protected final void createLocalDataset(String datasetName, Class<? extends Dataset> datasetClass) {
     createLocalDataset(datasetName, datasetClass, DatasetProperties.EMPTY);
   }
+
+  @Override
+  public void initialize(WorkflowContext context) throws Exception {
+    this.context = context;
+  }
+
+  @Override
+  public void destroy() {
+    // Nothing to do.
+  }
+
+  /**
+   * Return an instance of the {@link WorkflowContext} when the Workflow is running, otherwise return {@code null}.
+   */
+  protected final WorkflowContext getContext() {
```

Comment:
actually, we don't annotate the other ones as nullable.  Because we don't want people to call this when its not at runtime?